elastic/kibana
# [ResponseOps][Cases] Allow users to edit or delete template

## Summary

Merging into feature branch.

Implements edit and delete functionality https://github.com/elastic/kibana/issues/181864

Edit or delete template from case settings page.

https://github.com/elastic/kibana/assets/117571355/f6fd45fa-c1eb-45ab-8936-02c764dbadc4


**How to test**
- Go to Cases > Settings
- Add a template
- Click on edit template icon
- Update the fields
- Click save

- Click on delete template icon
- Click on confirm modal
- Click save

**Scenarios:** 
- Edit template with different custom fields
- Edit template  with connector
- Delete template

**Flaky Test** [here](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6331)

### Checklist
